agent/api/task/task.go

```diff
@@ -1450,6 +1464,68 @@ func (task *Task) addNetworkResourceProvisioningDependencyAwsvpc(cfg *config.Con
 	return nil
 }
 
+// addNetworkResourceProvisioningDependencyServiceConnectBridge creates one pause container per task container
+// except for SC container, and add a dependency for SC container to wait for all pause container running.
+//
+// Unlike AWSVPC pause containers that use CNI plugin for configuring netns when they go from RUNNING ->
+// RESOURCES_PROVISIONED, Bridge mode pause containers will have steady state RUNNING. All netns configuration and
+// CNI invocation will be performed solely by SC container.
+func (task *Task) addNetworkResourceProvisioningDependencyServiceConnectBridge(cfg *config.Config) error {
```

Comment:
There is lot of logic in here which is applicable only for Service Connect usecase. Could we move all the SC Logic to a new file? I believe that it would help in better structure and readability.

Reply:
Yeah! This [PR](https://github.com/aws/amazon-ecs-agent/pull/3211) starts the refactoring. I'll wait for it to be merged, and do a refactoring for my PR as well :)